src/metrics/categorical_accuracy.cpp

```diff
@@ -59,12 +59,33 @@ categorical_accuracy_metric::~categorical_accuracy_metric() {
 void categorical_accuracy_metric::setup(model& m) {
   metric::setup(m);
   const El::DistData dist_data(get_target_layer().get_prediction());
+  const El::Device dev = get_target_layer().get_prediction().GetLocalDevice();
   if (dist_data.colDist == El::MC
       && dist_data.rowDist == El::MR) {
-    m_prediction_values = new StarMRMat(*dist_data.grid, dist_data.root);
+    switch(dev) {
```

Comment:
We should just throw an error if the target layer is on GPU.

Reply:
Done